rasa/core/training/structures.py

```diff
@@ -392,6 +392,9 @@ def dump_to_file(self, filename, flat=False, e2e=False):
 
 
 class StoryGraph(object):
+    """The directed acyclic graph (DAG) of story-steps pooled from all the
+        stories present in the training data."""
```

Comment:
it's not acyclic, by default, there is a method that returns acyclic graph `with_cycles_removed`